app/code/Codifi/CustomerRequest/Model/ExportCsv.php

```diff
@@ -0,0 +1,181 @@
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        NoteRepository $noteRepository,
+        Csv $csvProcessor,
+        DirectoryList $directoryList,
+        FilterBuilder $filterBuilder,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        DateTime $dateTime
+    ) {
+        $this->fileSystem = $fileSystem;
+        $this->noteRepository = $noteRepository;
+        $this->csvProcessor = $csvProcessor;
+        $this->directoryList = $directoryList;
+        $this->filterBuilder = $filterBuilder;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->dateTime = $dateTime;
+    }
+
+    /**
+     * Pack customer notes to archive and delete from table
+     *
+     * @param int $period
+     * @return string
+     * @throws FileSystemException
+     * @throws NoSuchEntityException
+     */
+    public function export(int $period): string
+    {
+        $period = sprintf("-%s months", $period);
+        $newDate = $this->dateTime->date('Y-m-d H:m:s', strtotime($period));
+
+        $this->filterBuilder->setField('updated_at');
+        $this->filterBuilder->setConditionType('to');
+        $this->filterBuilder->setValue($newDate);
+        $filter = $this->filterBuilder->create();
+
+        $this->searchCriteriaBuilder->addFilter($filter);
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+
+        $noteList = $this->noteRepository->getList($searchCriteria);
+
+        $noteListItems = $noteList->getItems();
+
+        $content = [];
+        $isHeaderColsSet = false;
+        foreach ($noteListItems as $item) {
+            if (!$isHeaderColsSet) {
+                $content[] = array_keys($item->getData());
+                $isHeaderColsSet = true;
+            }
+            $content[] = $item->getData();
+        }
+
+        try {
+            $newDirectory = $this->fileSystem->getDirectoryWrite(DirectoryList::VAR_DIR);
+            $newDirectory->create(self::CUSTOMER_NOTE_ARCHIVE_PATH);
+        } catch (Exception $exception) {
+            throw $exception;
+        }
+
+        if ($newDirectory->isWritable(self::CUSTOMER_NOTE_ARCHIVE_PATH)) {
+            try {
+                $currentDateForName = $this->dateTime->date('Y_m_d');
+
+                $fileName = sprintf("customer_note_%s.csv", $currentDateForName);
+                $filePath = $this->directoryList->getPath(DirectoryList::VAR_DIR) .
+                    self::CUSTOMER_NOTE_ARCHIVE_PATH . $fileName;
+
+                $this->csvProcessor->setEnclosure('"');
+                $this->csvProcessor->setDelimiter(',');
+                $this->csvProcessor->saveData($filePath, $content);
+
+                $message = 'success';
+                foreach ($noteListItems as $item) {
+                    $noteId = $item->getNoteId();
+                    $this->noteRepository->deleteById($noteId);
+                }
+            } catch (FileSystemException $exception) {
```

Comment:
You should catch Exception or LocalizedException as it this case you can catch exception that related to delete data from repository functionality